### ISSUE TITLE
Movement

### ISSUE THREAD

**Issue by itajaja** [itajaja/sco#2]

**Comment by theodred25** [comment]
Great work! You also noticed there was no need for 24 time-steps in the example, since arc-nodes need not be counted separately (that was the redundancy I was thinking about this morning). 
There is, however, an important matter which we overlooked completely. A common situation is the following: I have 3 ships on planets A,B,C (I'm using the same map as in the example above), e.g. because they were produced on those planets, and I want to attack planet D; all the ships should arrive at the same time of course, but in the scenario descripted above I have no control over that.

**Comment by itajaja** [comment]
@theodred25 you are right about the missing scenario. This is easily doable in the following way. The player, when choosing what to move, organizes the ships into fleets, that are ought to travel together. All the ships in the fleet will move at the same speed, equal to the speed of the slowest ship in the fleet. Makes sense?

**Comment by lucaBuratti** [comment]
Dunno, Makes Sense to have fleets composed by ships that are in different places? I Think we have to decide which are the possibility of action we want to grant to the Player. If we want to introduce a concept of "fleet" that is different from the naive interpretation of "ships in The same place" we need to define it,its proprerties, And how it works

**Comment by itajaja** [comment]
Or maybe, for each movement, you can set up the speed (that of course cannot be higher than the max speed)?

**Comment by lucaBuratti** [comment]
Yeah setting the speed should resolve the problem. Anyway introduce the concept of fleet could extend the game. For the Player could be useful to Group ships in fleets in order to command them easily. What do You Think? A fleet Can be defined as a set of ships that has the movement speed equal to the slowest one. Is possible to add ships to a fleet if They are in The same position. The Player Can choose a Name for the fleet and Can command it like it was a single ship. This idea improves mainly the playability of the game. But i Think it shouldnt be only implemented as a utility of the user interface

**Comment by itajaja** [comment]
> This idea improves mainly the playability of the game. But i Think it shouldnt be only implemented as a utility of the user interface

yes agreed.

**Comment by theodred25** [comment]
I think the player should be able to decide both the speed and the sub-turns in which the ships depart. He could see a table like the one we drew, with all the sub-intervals. The reason is that if a number of ships share some of their path, but depart from different nodes, they should be able to traverse the same nodes at the same "time", since if an enemy fleet is waiting for them on one of those nodes, their effectiveness in combat is much worse if they arrive one at a time.

**Comment by Kimerao0** [comment]
I like the idea of giving the players as many option as possible, increasing the choices is always positive. The basic player would not mind the extra feature. Movimeng many units as one at the slowest speed is logical, but i approve the tabel like system also.

**Comment by lucaBuratti** [comment]
Let the player also decide in which subturn the ships decide to depart from, in my opinion adds to much complexity to the game. The mechanics of the game are public and if an expert player wants to exploit them at maximum can tune the velocity of the ships in order to coordinate them. I understand the scenario explained by @theodred25 but not allowing the player to choose in which subturn the ships depart is not a big limitation (in my opinion). How the movement is performed is an abstraction and the idea was to make it as a virtual-realTime. The input of the player is the destination and the speed. And this two inputs have an immediate interpretation. How the movement is actually performed should not be directly visible in game. An expert player can look for the mechanics explained in the wiki and then tune the speed to exploit them. And how is structured the game in this moment the input speed could be misleading, because changing the speed is not directly visible in the game to a player that does not know how the movement is performed but is intuitive, a ship A chases ship B, A attacks B only if is faster. The original idea of this abstraction was to let the player in the illusion of movement in real time.
